From e3c65a90327bf33a5cfa8bd6c29c822c6046e777 Mon Sep 17 00:00:00 2001
From: hyphop <art@khadas.com>
Date: Mon, 1 Jun 2020 15:00:42 +0900
Subject: [PATCH] amlogic_vim2_vim3_led_fix

add missed leds to VIM2
change VIM3 to GPIO_ACTIVE_HIGH + fix names and aliases

---
 .../dts/amlogic/meson-gxm-khadas-vim2.dts     |   14 +
 .../boot/dts/amlogic/meson-khadas-vim3.dtsi   |   12 +-
 2 files changed

diff --git a/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts b/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts
index 0686104c8..e7376fd05 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxm-khadas-vim2.dts
@@ -146,6 +146,18 @@
 		clock-frequency = <32768>;
 		pwms = <&pwm_ef 0 30518 0>; /* PWM_E at 32.768KHz */
 	};
+
+	leds {
+		compatible = "gpio-leds";
+		status = "okay";
+		led_white: led-white {
+			label = "vim:white";
+			gpios = <&gpio_ao GPIOAO_9 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "heartbeat";
+	};
+
+    };
+
 };
 
 &cec_AO {
@@ -390,3 +402,5 @@
 &usb0 {
 	status = "okay";
 };
+
+
diff --git a/arch/arm64/boot/dts/amlogic/meson-khadas-vim3.dtsi b/arch/arm64/boot/dts/amlogic/meson-khadas-vim3.dtsi
index 773816baf..d416d7def 100644
--- a/arch/arm64/boot/dts/amlogic/meson-khadas-vim3.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-khadas-vim3.dtsi
@@ -39,15 +39,15 @@
 	leds {
 		compatible = "gpio-leds";
 
-		led-white {
-			label = "vim3:white:sys";
-			gpios = <&gpio_ao GPIOAO_4 GPIO_ACTIVE_LOW>;
+		led_white: led-white {
+			label = "vim:white";
+			gpios = <&gpio_ao GPIOAO_4 GPIO_ACTIVE_HIGH>;
 			linux,default-trigger = "heartbeat";
 		};
 
-		led-red {
-			label = "vim3:red";
-			gpios = <&gpio_expander 5 GPIO_ACTIVE_LOW>;
+		led_red: led-red {
+			label = "vim:red";
+			gpios = <&gpio_expander 5 GPIO_ACTIVE_HIGH>;
 		};
 	};
 
-- 
2.17.1

